src/tslint.ts

```diff
@@ -91,6 +91,10 @@ class Linter {
             sourceFile = getSourceFile(this.fileName, this.source);
         }
 
+        if (sourceFile === undefined) {
+            throw new Error("Invalid source code: " + this.fileName);
```

Comment:
last time I touched this code IIRC `tsc` _does_ care about the file extension. having the wrong extension is a common error I see in issue reports so that's why I suggested it for a small improvement to the error message